autoscheduler/frontend/src/tests/redux/CourseCards.test.ts

```diff
@@ -465,60 +442,167 @@ describe('Course Cards Redux', () => {
       await new Promise(setImmediate);
 
       // assert
-      expect(store.getState().courseCards.numCardsCreated).toEqual(2);
-      expect(store.getState().courseCards[0].course).toEqual('CSCE 121');
-      expect(store.getState().courseCards[1].course).toEqual('MATH 151');
+      expect(store.getState().termData.courseCards.numCardsCreated).toEqual(2);
+      expect(store.getState().termData.courseCards[0].course).toEqual('CSCE 121');
+      expect(store.getState().termData.courseCards[1].course).toEqual('MATH 151');
+    });
+
+    describe("rejects an update when there's a term mismatch", () => {
+      test('when there are no cards and we call replaceCourseCards with a mismatched term', () => {
+        // arrange
+        const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+        store.dispatch(setTerm('202031'));
+
+        const courseCards: SerializedCourseCardOptions[] = [
+          {
+            course: 'CSCE 121',
+            customizationLevel: CustomizationLevel.BASIC,
+            collapsed: true,
+          },
+          {
+            course: 'MATH 151',
+            customizationLevel: CustomizationLevel.BASIC,
+            collapsed: false,
+          },
+        ];
+
+        // act
+        store.dispatch<any>(replaceCourseCards(courseCards, '201931'));
+
+        // assert
+        expect(store.getState().termData.courseCards.numCardsCreated).toEqual(1);
+        expect(store.getState().termData.courseCards[0].course).not.toEqual('CSCE 121');
+      });
+
+      test('with multiple cards for both right and mismatched terms', () => {
+        // arrange
+        const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+        store.dispatch(setTerm('202031'));
+
+        const wrongCourseCards: SerializedCourseCardOptions[] = [
+          {
+            course: 'CSCE 121',
+            customizationLevel: CustomizationLevel.BASIC,
+            collapsed: true,
+          },
+          {
+            course: 'MATH 151',
+            customizationLevel: CustomizationLevel.BASIC,
+            collapsed: false,
+          },
+        ];
+
+        const rightCourseCards: SerializedCourseCardOptions[] = [
+          {
+            course: 'COMM 203',
+            customizationLevel: CustomizationLevel.BASIC,
+            collapsed: true,
+          },
+          {
+            course: 'POLS 207',
+            customizationLevel: CustomizationLevel.BASIC,
+            collapsed: false,
+          },
+        ];
+
+        // act
+        store.dispatch<any>(replaceCourseCards(wrongCourseCards, '201931'));
+        store.dispatch<any>(replaceCourseCards(rightCourseCards, '202031'));
+
+        // assert
```

Comment:
This test doesn't check the number of course cards like the others you added

Reply:
Yeah that's true - but since both `201931` and `202031` have 2 course cards, the assertion that there are 2 cards would be pointless (since it would pass if the `201931` cards were used).